src/player.cpp

```diff
@@ -6445,6 +6447,9 @@ void player::hardcoded_effects(effect &it)
             }
         }
     } else if( id == effect_bleed ) {
+        // Prolonged haemorrhage is a significant risk for developing anaemia
+        vitamin_mod( vitamin_iron, rng( -1, -4 ) );
```

Comment:
Still, iron loss should be proportional to blood loss, not just time of bleeding. A gushing wound should "cost" more iron than a trickle. The iron decrement should be moved below, inside the `if`.